### PR TITLE
Add option for cert syncer to write a file on caches synced that can be used for readiness probe

### PR DESCRIPTION
**Pull request by tnozicka**
We will use it for readiness probe.

I've tested it in my dev branch, but I'll prepare fake bumps before we merge.
/hold

/cc @deads2k @soltysh @mfojtik

### CODE REVIEW

**Comment by tnozicka** [comment]
kube-apiserver-operator in action https://github.com/openshift/cluster-kube-apiserver-operator/pull/662

File: pkg/operator/staticpod/certsyncpod/certsync_controller.go [hunk 1]
```diff
@@ -298,6 +300,21 @@ func (c *CertSyncController) Run(workers int, stopCh <-chan struct{}) {
...
+	if len(c.cachesSyncedFile) != 0 {
+		// Signal caches sync using a file (e.g. for readiness probe)
+		f, err := os.OpenFile(c.cachesSyncedFile, os.O_APPEND|os.O_CREATE|os.O_RDONLY, 0600)
+		if err != nil {
+			klog.Fatal(err)
+		}
+
+		klog.V(1).Infof("CertSyncer caches synced file %q created.", c.cachesSyncedFile)
+
+		err = f.Close()
```
Comment (mfojtik) [inline pkg/operator/staticpod/certsyncpod/certsync_controller.go]: defer f.Close()

Comment (mfojtik) [reply pkg/operator/staticpod/certsyncpod/certsync_controller.go]: actually, nvmd.

Comment (tnozicka) [reply pkg/operator/staticpod/certsyncpod/certsync_controller.go]: you don't check the error that way (unless you wrap it in a lambda) and there isn't a place for branching in between to escape the flow

**Comment by tnozicka** [comment]
https://github.com/openshift/cluster-kube-apiserver-operator/pull/662 si green

the file seems to be working
```
2019-11-22T13:00:49.717350963Z I1122 13:00:49.716220       1 certsync_controller.go:294] Starting CertSyncer
2019-11-22T13:00:49.717350963Z I1122 13:00:49.716383       1 observer_polling.go:137] Starting file observer
2019-11-22T13:00:55.317438531Z I1122 13:00:55.317385       1 certsync_controller.go:301] CertSyncer caches synced
2019-11-22T13:00:55.317776101Z I1122 13:00:55.317744       1 certsync_controller.go:310] CertSyncer caches synced file "/var/run/informer-caches-synced" created.
2019-11-22T13:00:55.317953398Z I1122 13:00:55.317872       1 certsync_controller.go:84] Syncing configmaps: [{aggregator-client-ca false} {client-ca false} {trusted-ca-bundle true}]
`
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/662/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws/3008/artifacts/e2e-aws/must-gather/registry-svc-ci-openshift-org-ci-op-lwp8pzd2-stable-sha256-64c63eedf863406fbc6c7515026f909a7221472cf70283708fb7010dd5e6139e/namespaces/openshift-kube-apiserver/pods/kube-apiserver-ip-10-0-133-197.ec2.internal/kube-apiserver-cert-syncer-6/kube-apiserver-cert-syncer-6/logs/current.log

**Comment by deads2k** [comment]
When a pod crash loops for any reason on a node, the pod starts back up and gets the same empty dirs, does it also get the same layers where it wrote files?

This is really dirty way to get the functionality.  I could live it (if it works on crashloops) for the purposes of a backport, but it needs to become a real healthz endpoint in the end.

**Pull request closed without merging**